From 72f9b05cd93e09b2e881af2112149ce979fc64ad Mon Sep 17 00:00:00 2001
From: Bin Jiang <bin.jiang@windriver.com>
Date: Tue, 28 Jan 2014 17:50:08 +0800
Subject: [PATCH 809/838] powerpc/kgdb: Fix gdb fails to get PC register in
 e500mc

gdb can't single-step since it fails to get PC.

kgdb send message including PC to gdb, but the address to store PC is
wrong. So gdb fails to get it.

In e500mc, each FP register is 64-bit nor 32-bit which is only used for
e500 SPE.
In message sent to gdb, the PC is stored behind FP. So the wrong size of
FP makes PC stored in wrong address.

Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/powerpc/kernel/kgdb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/kernel/kgdb.c b/arch/powerpc/kernel/kgdb.c
index 42001ff..e3f83eb 100644
--- a/arch/powerpc/kernel/kgdb.c
+++ b/arch/powerpc/kernel/kgdb.c
@@ -264,7 +264,7 @@ void sleeping_thread_to_gdb_regs(unsigned long *gdb_regs, struct task_struct *p)
 #define GDB_SIZEOF_REG sizeof(unsigned long)
 #define GDB_SIZEOF_REG_U32 sizeof(u32)
 
-#ifdef CONFIG_FSL_BOOKE
+#if defined(CONFIG_FSL_BOOKE) && defined(CONFIG_SPE)
 #define GDB_SIZEOF_FLOAT_REG sizeof(unsigned long)
 #else
 #define GDB_SIZEOF_FLOAT_REG sizeof(u64)
-- 
2.0.2

